syft/generic/frameworks/hook/hook_args.py

```diff
@@ -320,10 +321,12 @@ def build_unwrap_args_with_rules(args, rules, return_tuple=False):
     lambdas = [
         typed_identity(a)  # return the same obj with an identity fct with a type check if needed
         if not r  # if the rule is a number == 0.
+        else build_unwrap_args_with_rules(a, r, True, True)
```

Comment:
Since the `return_list` flag causes the function to coerce the return type to `list`, could we do without the flag and make this line something like:
```suggestion
        else list(build_unwrap_args_with_rules(a, r, True))
```
If so, the lines below could also be removed.